pkgs/by-name/ya/yaak/package.nix

```diff
@@ -54,31 +55,44 @@ rustPlatform.buildRustPackage (finalAttrs: {
     makeWrapper
   ];
 
-  buildInputs = [
-    glib
-    gtk3
-    openssl
-    webkitgtk_4_1
-    pango
-    cairo
-    pixman
-  ];
+  buildInputs =
+    [
+      glib
+      gtk3
+      openssl
+      pango
+      cairo
+      pixman
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      webkitgtk_4_1
+    ];
 
   env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
 
   postPatch = ''
     substituteInPlace src-tauri/tauri.conf.json \
-      --replace-fail '"createUpdaterArtifacts": "v1Compatible"' '"createUpdaterArtifacts": false'
+      --replace-fail '"createUpdaterArtifacts": "v1Compatible"' '"createUpdaterArtifacts": false' \
+      --replace-fail '"0.0.0"' '"${finalAttrs.version}"'
     substituteInPlace package.json \
       --replace-fail '"bootstrap:vendor-node": "node scripts/vendor-node.cjs",' "" \
       --replace-fail '"bootstrap:vendor-protoc": "node scripts/vendor-protoc.cjs",' ""
   '';
 
+  archPlatforms =
```

Comment:
Maybe move `archPlatforms` into a `let ... in` block, as it's not a valid attribute of mkDerivation?

```
preBuild =
  let
    archPlatforms = ... 
  in
  ''
    mkdir -p ...
  ''
```